### ISSUE TITLE
Add list view

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#712]
**Describe the solution you'd like**
![image](https://user-images.githubusercontent.com/11173328/119218800-31701d00-bada-11eb-9cd0-d444238f783a.png)

Connect to the get all books endpoint on the backend and populate a table as shown above (we can ignore the thumbnails for now).

Focus on implementing the part in the blue box.

This should be responsive.

**Additional context**
Branch from `0.2.0` and create a pull request to go back into it.

**Comment by Anu-123-gif** [comment]
Hey @knjk04 I would like to work on this issue , with some guidance if possible :)
Meanwhile I'll start searching of possible solutions and discuss them here as soon as possible.

**Comment by knjk04** [comment]
Thanks!

**Comment by knjk04** [comment]
@Anu-123-gif How are you getting on with this?

**Comment by Anu-123-gif** [comment]
Hey @knjk04 I have understood how to approach the issue. I have two doubts:
-> In the backend, where can I find the endpoints using which i'll populate the list,
-> When I open the frontend in IntelliJ IDEA I am shown ' *.ts files are supported by IntelliJ IDEA Ultimate' , do I have to switch that instead?

Thank you for your time :)

**Comment by knjk04** [comment]
> In the backend, where can I find the endpoints using which i'll populate the list,
See https://github.com/Project-Books/book-project/blob/0.2.0/backend/src/main/java/com/karankumar/bookproject/backend/controller/BookController.java#L107. It's a GET on [localhost]:/api/books. 

> When I open the frontend in IntelliJ IDEA I am shown ' *.ts files are supported by IntelliJ IDEA Ultimate' , do I have to switch that instead?

Do you have IntelliJ IDEA Ultimate? If not, you may want to open the frontend in something else (e.g. Visual Studio Code)

**Comment by Anu-123-gif** [comment]
Thanks a lot for your help! 
I shifted to VS code and it works perfectly there.
I was shown an error `'--jsx' is not set`, to solve this I had to change [https://github.com/Project-Books/book-project/blob/main/frontend/tsconfig.json](url) . I changed `"jsx": "react-jsx"` to `"jsx": "react"` here. Is there something else I should've done?

**Comment by knjk04** [comment]
You're welcome! The link takes me to a no results page, unfortunately

**Comment by Anu-123-gif** [comment]
Sorry about that, here is the working link https://github.com/Project-Books/book-project/blob/main/frontend/tsconfig.json

**Comment by knjk04** [comment]
No matter. I haven't needed to change that setting myself, but the change you made is fine if it works for us (we'll see when reviewing your PR)

**Comment by Anu-123-gif** [comment]
Hey @knjk04 ,
I am trying to update my fork but its only updating the main branch, can you please help me access the 0.2.0 branch?
Thanks!

**Comment by knjk04** [comment]
Hi @Anu-123-gif,

I assumed you forked, not cloned, the repository. Have you tried switching to the 0.2.0 with `git checkout`?
`$ git checkout -b 0.2.0`
`$ git pull origin 0.2.0`

**Comment by Anu-123-gif** [comment]
Thanks!
There was some issue with my fork it seems ,I deleted the fork and forked the repo again.

**Comment by knjk04** [comment]
@Anu-123-gif How are you getting on with this?

**Comment by knjk04** [comment]
@Anu-123-gif Unassigning due to no response

**Comment by rhit-namburs** [comment]
Hi I would like to work on this issue!

**Comment by knjk04** [comment]
Hi @rhit-namburs, thanks for your interest in this issue. Please note that, as per our contributing guidelines, we only assign at most issue at a time to new contributors

**Comment by Pranjal-dew** [comment]
hi there, can I contribute to this issue?

**Comment by RivaD2** [comment]
@knjk04 Hey there!

I will take this one on 👍

**Comment by knjk04** [comment]
@RivaD2 Hi Riva, thanks!

**Comment by RivaD2** [comment]
@knjk04 I have started working on this. How will the user get to the list view? Once the user logs in, they land on `My Books` with the shelf view.

**Comment by knjk04** [comment]
@RivaD2 There'll be a toggle at the bottom (see the screenshot in the issue description). There's a separate issue for this (#679)

**Comment by RivaD2** [comment]
@knjk04, thank you!

 I believe that the toggle functionality should be built first, then I could move on to this ticket. Can you assign me #679?
